src/sentry/services/hybrid_cloud/auth/__init__.py

```diff
@@ -134,6 +134,16 @@ def get_auth_providers(self, organization_id: int) -> List[ApiAuthProvider]:
         """
         pass
 
+    @abc.abstractmethod
+    def handle_new_membership(
+        self,
+        request: Request,
```

Comment:
Hmm, my worry here is the requirement of the request object.  Is there, any way we can further unpack this?  Because when this becomes an RPC invocation, we don't want to be serializing an entire request object.

Reply:
It looks like we rely on two session values, `invite_token` and `invite_member_id` and the request user.

Reply:
Sounds good, but it would cascade some sizeable changes through `ApiInviteHelper`. How about as a separate PR?

Reply:
Totally makes sense to do separately :+1:

Reply:
https://github.com/getsentry/sentry/pull/43211